src/services/refactors/addOrRemoveBracesToArrowFunction.ts

```diff
@@ -36,6 +36,24 @@ namespace ts.refactor.addOrRemoveBracesToArrowFunction {
         }];
     }
 
+    /**
+     * used to check if the last character in the returnStatement is a semicolon
+     */
+    function hasSemiColon(returnStatement: ReturnStatement) {

```

Comment:
I'm not sure which semicolon we want to keep here, but `//2` should already be preserved. Either way there is a `SyntaxKind.SemicolonToken`.

```ts
const b = (a: number) => { 
    return { a: a }; //1
}; //2
```

Reply:
It's hard to remember why this was happening, but if there is a semicolon, it wasn't being included as part of the refactor when grabbing the trailing comments, hence why I added this. 

If I jog my memory, I think it was for the semicolon that come at the end of the function body but before the trailing comment:
```js
const a = (a: number) => a; /* trailing */
```

I'm not sure what action you want me to take.

Reply:
Sorry, my comment was definitely vague.

I wouldn't worry about trying to copy the semicolon here. If the original has two semicolons we will already end up with the semicolon from the original variable statement, assuming we just are just replacing the function body:

```ts
const b = (a: number) => { 
    return a; 
};|<
```

```ts
const b = (a: number) => a;|<
```

No semicolons gives no semicolons

```ts
const b = (a: number) => { 
    return a
}
```

```ts
const b = (a: number) => a
```

And then we have two cases of inconsistent semicolons
```ts
const b = (a: number) => { 
    return a
};
```

```ts
const b = (a: number) => { 
    return a;
}
```

Which do we pick? I don't think we need to think too hard about making complex rules to decide which one to use. If the original code has inconsistent semicolon usage, we don't need to be consistent in providing/not providing a semicolon.